Box2D/CMakeLists.txt

```diff
@@ -0,0 +1,25 @@
+cmake_minimum_required(VERSION 3.6)
+
+set(lib_name box2d)
```

Comment:
1. Why we need both lib_name and target_name?
Why not lib_name == target_name?
2. project should be second line after cmake_minimum_required
project(...) define variable PROJECT_NAME which can be used
https://cmake.org/cmake/help/latest/variable/PROJECT_NAME.html#variable:PROJECT_NAME
3. I think that we should use target_include_directories, target_link_libraries instead of set_target_properties
https://stackoverflow.com/questions/36648375/what-are-the-differences-between-imported-target-and-interface-libraries/45212811#45212811

I propose to simplify it to (here and in other places)
```
cmake_minimum_required(VERSION 3.6)
project(box2d)

include(../cmake/CocosExternalConfig.cmake)
add_library(${PROJECT_NAME} STATIC IMPORTED GLOBAL)

target_include_directories(${PROJECT_NAME} INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}/include)

if(WINDOWS)
    target_link_libraries(${PROJECT_NAME} INTERFACE "${platform_spec_path}/lib${PROJECT_NAME}.lib")
else()
    target_link_libraries(${PROJECT_NAME} INTERFACE "${platform_spec_path}/lib${PROJECT_NAME}.a")
  )
endif()
```

Reply:
> Why we need both lib_name and target_name?
with `ext_` prefix, I know this libs clearly comes from cocos2d-x/external, not a system libs

> define variable PROJECT_NAME which can be used inside CMakeLists.txt

if the CMakeLists.txt can't been consider as a single project, it must be used as a submodule  into a cmake project. so what's the effect to define it?

> I think that we should use target_include_directories, target_link_libraries instead of set_target_properties

we can't use `target_include_directories` and `target_link_libraries` for __improted target__ on CMake 3.6, or you will get errors.

Reply:
But still why not lib_name == target_name e.g. ext_box2d

Reply:
firstly, I want to say both is OK to aim to same logic. 

As I considered, the variable name will indicate the meanings, help us to understand. the library name of __box2d__ is box2d, so I set `lib_name` to `box2d`. I don't think uniform both variable into one is meaningful.